misc/language.c

```diff
@@ -0,0 +1,839 @@
+    /*0614*/ { "na" ,  300 },
+    /*0615*/ { "nb" ,  312 },
+    /*0616*/ { "nd" ,  303 },
+    /*0617*/ { "ne" ,  306 },
+    /*0618*/ { "ng" ,  304 },
+    /*0619*/ { "nl" ,  115 },
+    /*0620*/ { "nn" ,  311 },
+    /*0621*/ { "no" ,  315 },
+    /*0622*/ { "nr" ,  302 },
+    /*0623*/ { "nv" ,  301 },
+    /*0624*/ { "ny" ,  320 },
+    /*0625*/ { "oc" ,  325 },
+    /*0626*/ { "oj" ,  326 },
+    /*0627*/ { "om" ,  328 },
+    /*0628*/ { "or" ,  327 },
+    /*0629*/ { "os" ,  330 },
+    /*0630*/ { "pa" ,  337 },
+    /*0631*/ { "pi" ,  344 },
+    /*0632*/ { "pl" ,  345 },
+    /*0633*/ { "ps" ,  350 },
+    /*0634*/ { "pt" ,  347 },
+    /*0635*/ { "qu" ,  351 },
+    /*0636*/ { "rm" ,  356 },
+    /*0637*/ { "rn" ,  359 },
+    /*0638*/ { "ro" ,  358 },
+    /*0639*/ { "ru" ,  361 },
+    /*0640*/ { "rw" ,  224 },
+    /*0641*/ { "sa" ,  368 },
+    /*0642*/ { "sc" ,  400 },
+    /*0643*/ { "sd" ,  393 },
+    /*0644*/ { "se" ,  386 },
+    /*0645*/ { "sg" ,  363 },
+    /*0646*/ { "si" ,  379 },
+    /*0647*/ { "sk" ,  383 },
+    /*0648*/ { "sl" ,  384 },
+    /*0649*/ { "sm" ,  390 },
+    /*0650*/ { "sn" ,  392 },
+    /*0651*/ { "so" ,  396 },
+    /*0652*/ { "sq" ,   12 },
+    /*0653*/ { "sr" ,  402 },
+    /*0654*/ { "ss" ,  405 },
+    /*0655*/ { "st" ,  398 },
+    /*0656*/ { "su" ,  407 },
+    /*0657*/ { "sv" ,  411 },
+    /*0658*/ { "sw" ,  410 },
+    /*0659*/ { "ta" ,  416 },
+    /*0660*/ { "te" ,  418 },
+    /*0661*/ { "tg" ,  422 },
+    /*0662*/ { "th" ,  424 },
+    /*0663*/ { "ti" ,  427 },
+    /*0664*/ { "tk" ,  439 },
+    /*0665*/ { "tl" ,  423 },
+    /*0666*/ { "tn" ,  437 },
+    /*0667*/ { "to" ,  434 },
+    /*0668*/ { "tr" ,  442 },
+    /*0669*/ { "ts" ,  438 },
+    /*0670*/ { "tt" ,  417 },
+    /*0671*/ { "tw" ,  445 },
+    /*0672*/ { "ty" ,  414 },
+    /*0673*/ { "ug" ,  449 },
+    /*0674*/ { "uk" ,  450 },
+    /*0675*/ { "ur" ,  453 },
+    /*0676*/ { "uz" ,  454 },
+    /*0677*/ { "ve" ,  456 },
+    /*0678*/ { "vi" ,  457 },
+    /*0679*/ { "vo" ,  458 },
+    /*0680*/ { "wa" ,  466 },
+    /*0681*/ { "wo" ,  467 },
+    /*0682*/ { "xh" ,  469 },
+    /*0683*/ { "yi" ,  472 },
+    /*0684*/ { "yo" ,  473 },
+    /*0685*/ { "za" ,  478 },
+    /*0686*/ { "zh" ,   78 },
+    /*0687*/ { "zu" ,  480 },
+    /*0688*/ { "in" ,  195 }, /* deprecated */
+    /*0689*/ { "iw" ,  172 }, /* deprecated */
+    /*0690*/ { "ji" ,  472 }, /* deprecated */
+    /*0691*/ { "jw" ,  202 }, /* deprecated */
+    /*0692*/ { "mo" ,  358 }, /* deprecated */
+    /*0693*/ { "sh" ,  693 }, /* deprecated (no equivalent) */
+    { "", 0 }
+};
+
+static int lang_table_compare(const void *lhs, const void *rhs)
+{
+    return strcmp(lhs, ((const LangEntry *)rhs)->str);
+}
+
+/**
+ * Convert a language code to a target codespace. The source codespace is guessed.
+ * @return NULL if the provided lang is null or invalid.
+ */
+static const char *ff_convert_lang_to(const char *lang, enum AVLangCodespace target_codespace)
```

Comment:
since we only ever use this with `target==AV_LANG_ISO639_2_BIBL` wouldn't keeping (most) of the old code been fine?
(it has a direct and smaller mapping)

Reply:
I didn't validate exact mappings, but ffmpeg's one looks more complete. Also we use `target==AV_LANG_ISO639_2_BIBL`, but for source we can use any. But sure old list is good enough. I can restore it.

Reply:
Old code need fixing tho. `sizeof(key->match)` is not proper way to get length of string, it is sizeof array of 4 elements... Well yeah, it doesn't work. One of the reasons it was broken...

Reply:
Done. I restored the table, but rewritten the rest of the code.